packages/test/test-utils/src/containerUtils.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IContainer } from "@fluidframework/container-definitions";
+import { ConnectionState, Container } from "@fluidframework/container-loader";
+import { PromiseExecutor, timeoutPromise, TimeoutWithError } from "./timeoutUtils";
+
+/**
+ * Waits for the specified container to emit a 'connected' event.
+ *
+ * @deprecated Use waitForContainerConnection instead.
+ * Note that an upcoming release will change the default parameters on that function to:
+ * - failOnContainerClose = true
+ * - timeoutOptions.durationMs = 1s
+ */
+export async function ensureContainerConnected(container: Container): Promise<void> {
```

Comment:
Is this used outside our repo? I don't think so but I guess its better to be on the cautious side and have this in place.

Maybe we need the ability to mark these packages as internal and not provide back-compat guarantees.

Reply:
Yeah, I brought up this PR in office hours yesterday to see if we're making the same back-compat guarantees on our test-related packages and we agreed we do, thus this refactor.

And we also agreed that there's probably room for improvement, mainly along the lines of cleaning up the current API surface, and somehow being able to differentiate between things we export for general use and things we export but just for internal purposes.

But for now, back-compat it is. Once this merges and integrates into next, I'll open another PR to remove it in the next major version.